src/db/add_package.rs

```diff
@@ -316,26 +317,42 @@ fn add_keywords_into_database(
     pkg: &MetadataPackage,
     release_id: i32,
 ) -> Result<()> {
-    for keyword in &pkg.keywords {
-        let slug = slugify(&keyword);
-        let keyword_id: i32 = {
-            let rows = conn.query("SELECT id FROM keywords WHERE slug = $1", &[&slug])?;
-            if !rows.is_empty() {
-                rows[0].get(0)
-            } else {
-                conn.query(
-                    "INSERT INTO keywords (name, slug) VALUES ($1, $2) RETURNING id",
-                    &[&keyword, &slug],
-                )?[0]
-                    .get(0)
-            }
-        };
+    let wanted_keywords: HashMap<String, String> = pkg
+        .keywords
+        .iter()
+        .map(|kw| (slugify(&kw), kw.clone()))
+        .collect();
+
+    let existing_keyword_slugs: HashSet<String> = conn
+        .query(
+            "SELECT slug FROM keywords WHERE slug = ANY($1)",
```

Comment:
Hmm, shouldn't this be `WHERE slug IN $1`? My understanding is that `ANY($1)` does a truthiness comparison and return TRUE if any of them are non-empty.

Reply:
Now trying to remember my decision process all these weeks ago :) 

I remember `slug IN $1` didn't work here in combination with binding a array to `$1`. 

I'll check my logic in the PR again to see if it really behaves like I thought it does, or it does like you described.

Reply:
According to postgres docs

> `IN` is equivalent to `= ANY`

but that's definitely not what I've found in practice with array-typed values, those have always needed `= ANY` 🤷.

Reply:
so, I checked again. the `field = ANY($array)` is what we need here, when binding an array. 

( I saw the comment by @Nemo157 after I did my test, so there is no new information, but here is the test anyways: 
```
psql (13.3)

cratesfyi=# select * from keywords;
┌────┬────────────────┬────────────────┐
│ id │      name      │      slug      │
├────┼────────────────┼────────────────┤
│ 24 │ pool           │ pool           │
│ 25 │ database       │ database       │
│ 26 │ iterator       │ iterator       │
│ 27 │ data-structure │ data-structure │
│ 28 │ zip            │ zip            │
│ 29 │ product        │ product        │
│ 30 │ group-by       │ group-by       │
└────┴────────────────┴────────────────┘
(7 rows)

cratesfyi=# select * from keywords where name = ANY(ARRAY['pool', 'zip']);
┌────┬──────┬──────┐
│ id │ name │ slug │
├────┼──────┼──────┤
│ 24 │ pool │ pool │
│ 28 │ zip  │ zip  │
└────┴──────┴──────┘
(2 rows)

cratesfyi=# select * from keywords where name = ALL(ARRAY['pool', 'zip']);
┌────┬──────┬──────┐
│ id │ name │ slug │
├────┼──────┼──────┤
└────┴──────┴──────┘
(0 rows)

cratesfyi=#
```

Reply:
I am not sure of the state of this; was there a bug originally? If so, is it fixed?

Reply:
You mean a bug in the PR here? No, this works as it should, though the `name = ANY(` thing looks confusing. 

( there are also tests covering this now)